packages/ckeditor5-engine/tests/model/utils/insertcontent.js

```diff
@@ -1299,128 +1302,1629 @@ describe( 'DataController utils', () => {
+						'marker-a': { start: [ 0 ], end: [ 0 ] }
+					}, [ 1 ] );
+
+					const expectedMarker = model.markers.get( 'marker-a' );
+
+					expect( getData( model ) ).to.equal( '<paragraph>foo[]</paragraph><paragraph><imageInline></imageInline></paragraph>' );
+
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 1 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 1 ] );
+				} );
+
+				it( 'should create collapsed marker after inserting imageInline before paragraph', () => {
+					// []<paragraph>foo</paragraph>
+					// {}<imageInline></imageInline>
+					//
+					// {}<paragraph><imageInline></imageInline></paragraph><paragraph>foo</paragraph>
+					setData( model, '[]<paragraph>foo</paragraph>' );
+
+					insertHelper( '<imageInline></imageInline>', {
+						'marker-a': { start: [ 0 ], end: [ 0 ] }
+					}, [ 0 ] );
+
+					const expectedMarker = model.markers.get( 'marker-a' );
+
+					expect( getData( model ) ).to.equal( '<paragraph><imageInline></imageInline></paragraph><paragraph>[]foo</paragraph>' );
+
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 0 ] );
+				} );
+
+				it( 'should create collapsed marker after inserting imageBlock at the beginning of paragraph', () => {
+					// <paragraph>[]foo</paragraph>
+					// {}<imageBlock></imageBlock>
+					//
+					// {}<imageBlock></imageBlock><paragraph>foo</paragraph>
+					setData( model, '<paragraph>[]foo</paragraph>' );
+
+					insertHelper( '<imageBlock></imageBlock>', {
+						'marker-a': { start: [ 0 ], end: [ 0 ] }
+					} );
+
+					const expectedMarker = model.markers.get( 'marker-a' );
+
+					expect( getData( model ) ).to.equal( '[<imageBlock></imageBlock>]<paragraph>foo</paragraph>' );
+
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 0 ] );
+				} );
+
+				it( 'should create collapsed marker after inserting imageBlock in the middle of paragraph', () => {
+					// <paragraph>f[]oo</paragraph>
+					// {}<imageBlock></imageBlock>
+					//
+					// <paragraph>f{}</paragraph><imageBlock></imageBlock><paragraph>oo</paragraph>
+					setData( model, '<paragraph>f[]oo</paragraph>' );
+
+					insertHelper( '<imageBlock></imageBlock>', {
+						'marker-a': { start: [ 0 ], end: [ 0 ] }
+					} );
+
+					const expectedMarker = model.markers.get( 'marker-a' );
+
+					expect( getData( model ) ).to.equal( '<paragraph>f</paragraph>[<imageBlock></imageBlock>]<paragraph>oo</paragraph>' );
+
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0, 1 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 0, 1 ] );
+				} );
+
+				it( 'should create collapsed marker after inserting imageBlock at the end of paragraph', () => {
+					// <paragraph>foo[]</paragraph>
+					// {}<imageBlock></imageBlock>
```

Comment:
Interesting that in this case it was moved out of the paragraph, but in previous case it stayed in the paragraph.

Reply:
It's since in the previous example the insertion of `imageBlock` split the original paragraph into two and `affectedRange` starts from the beginning of the first split paragraph.  
  
In that case the `imageBlock` element is inserted just after the paragraph even if the selection is inside.